### PR TITLE
fix: apply prettier config at the very end to disable all required rules

### PR DESCRIPTION
**Pull request by grabbou**
Prettier should be applied as a very last rule to make sure all rules (such as ones defined by `typescript-eslint`) are disabled and do not cause issues.

**Context:**

Prettier ESLint plugin includes an ESLint config that disables all potentially conflicting ESLint rules that are already handled by Prettier. A great example is `@typescript-eslint/no-extra-semi`. 

ESLint allows to extend multiple configs via `extends` array. We need to make sure Prettier config is the very last config in that array, otherwise some plugins may accidentally re-enable conflicting rules.

A great example is when Typescript ESLint config is loaded after Prettier config and it re-enables some rules, such as `@typescript-eslint/no-extra-semi`. You can see the difference in a configuration snapshot after changing the order of elements in the array.

### CODE REVIEW

File: __snapshots__/preset.test.ts.snap [hunk 1]
```diff
@@ -1097,11 +1097,6 @@ exports[`should have a correct configuration for a TypeScript file 1`] = `
       "error",
       {
         "paths": [
-          {
-            "allowTypeImports": true,
-            "message": "Please only use Providers instantiated in constants/providers to improve traceability.",
-            "name": "@ethersproject/providers",
-          },
```
Comment (grabbou) [inline __snapshots__/preset.test.ts.snap:1104]: chore: looks like some snapshots are outdated on main

**Comment by zzmp** [comment]
I can't tell what this accomplishes, or what your comment means. Could you give more details in the PR description?

**Comment by grabbou** [comment]
Updated PR description, hope it's a bit clearer now!

File: __snapshots__/preset.test.ts.snap [hunk 2]
```diff
@@ -88,7 +88,7 @@ exports[`should have a correct configuration for a Cypress e2e file 1`] = `
       "off",
     ],
     "@typescript-eslint/no-extra-semi": [
-      "error",
```
Comment (zzmp) [inline __snapshots__/preset.test.ts.snap:91]: Shouldn't this be error? Will the prettier config still be applied to cypress/jest files?

**Review by zzmp — APPROVED** [review]

**Pull request merged**